### PR TITLE
Improvements of boxed correlation sum

### PR DESCRIPTION
**Pull request by Datseris**
@apbraun let's do all necessary improvements in this PR.

### CODE REVIEW

File: src/dimensions/correlationdim.jl [hunk 1]
```diff
@@ -230,11 +231,16 @@ function boxed_correlationsum(
...
+# An algorithm to find the ideal choice of a prism size for the boxed correlation dimension.
+# See Theiler as mentioned in `boxed_correlationsum`
 function _autoprismdim(data)
     D = dimension(data)
-    L = length(data)
-    # The best choice given by Theiler is $P = 0.5\log_2N$ and should be used when $D$ exceeds $0.75\log_2N$.
-    return size(data, 2)
+    N = length(data)
+    if D > 0.75 * log2(N)
+        return ceil(0.5 * log2(N))
```
Comment (Datseris) [inline src/dimensions/correlationdim.jl:240]: Should we have this be inside a `max(2, ...)` ? I think we say in the paper someone used always 2?

Comment (apbraun) [reply src/dimensions/correlationdim.jl:240]: Yes that seems reasonable.

**Comment by Datseris** [comment]
@apbraun can you please remind me: these are all the improvements we wanted to do, right? I can merge away?

**Comment by apbraun** [comment]
Yes these are all improvements. We spoke about adjusting the `_autoprism` algorithm to the Theiler version and catch `r0` < `ε0` in the `boxed_correlationsum`, I also spotted another bug in `boxed_correlationsum` that I fixed, so from my part we're ready to go.

**Comment by Datseris** [comment]
@apbraun please give me some help here. In our paper you said that "Bueno-Orovio and Pérez-García chose P=2 for D>2" where I assume that P stands for the Prism dimension. Here we don't do that, but we use I think an older approach by Theiler. Which one should we use in the end...?

**Pull request merged**